src/mlpack/methods/ann/layer/minibatch_discrimination_impl.hpp

```diff
@@ -0,0 +1,143 @@
+/**
+ * @file minibatch_discrimination_impl.hpp
+ * @author Saksham Bansal
+ *
+ * Implementation of the MiniBatchDiscrimination layer class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_MINIBATCH_DISCRIMINATION_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_MINIBATCH_DISCRIMINATION_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "minibatch_discrimination.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+MiniBatchDiscrimination<InputDataType, OutputDataType
+>::MiniBatchDiscrimination() :
+  A(0),
+  B(0),
+  C(0),
+  batchSize(0)
+{
+  // Nothing to do here.
+}
+
+template <typename InputDataType, typename OutputDataType>
+MiniBatchDiscrimination<InputDataType, OutputDataType
+>::MiniBatchDiscrimination(
+    const size_t inSize,
+    const size_t outSize,
+    const size_t features) :
+    A(inSize),
+    B(outSize - inSize),
+    C(features),
+    batchSize(0)
+{
+  weights.set_size(A * B * C, 1);
+}
+
+template<typename InputDataType, typename OutputDataType>
+void MiniBatchDiscrimination<InputDataType, OutputDataType>::Reset()
+{
+  weight = arma::mat(weights.memptr(), B * C, A, false, false);
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename eT>
+void MiniBatchDiscrimination<InputDataType, OutputDataType>::Forward(
+    const arma::Mat<eT>&& input, arma::Mat<eT>&& output)
+{
+  batchSize = input.n_cols;
+  tempM = weight * input;
+  M = arma::cube(tempM.memptr(), B, C, batchSize, false, false);
+  distances.set_size(B, batchSize, batchSize);
+  output.set_size(B, batchSize);
+  for (size_t i = 0; i < M.n_slices; i++)
```

Comment:
I would add a blank line between the `for` loop and above code for aesthetics, but that's just me being extreme OCD.